Src/Support/Google.Apis.Auth/OAuth2/ServiceCredential.cs

```diff
@@ -173,9 +173,16 @@ public ServiceCredential(Initializer initializer)
         /// <inheritdoc/>
         public async Task InterceptAsync(HttpRequestMessage request, CancellationToken cancellationToken)
         {
-            var accessToken = await GetAccessTokenWithHeadersForRequestAsync(request.RequestUri.AbsoluteUri, cancellationToken).ConfigureAwait(false);
+            var accessToken = await GetAccessTokenWithHeadersForRequestAsync(GetAudience(request.RequestUri), cancellationToken).ConfigureAwait(false);
```

Comment:
What's the purpose of this change? It seems unrelated to adding a signing key to the JWT.

Reply:
It's described on the second line on the commit description.
If we pass the whole URL as it was being done before the token won't work, the audience needs to be just the https://service_name part. gRPC was passing the correct value to GetAccessToken... but Apiary (the interceptor) wasn't. This is unrelated to adding the kid but it was a bug found when I added the tests.

Reply:
Ah, sorry about that - missed it. Goodo.